src/frontend/Util_ROM.cpp

```diff
@@ -422,16 +503,74 @@ int Reset()
     }
     else
     {
-        SetupSRAMPath(0);
-        if (!NDS::LoadROM(ROMPath[ROMSlot_NDS], SRAMPath[ROMSlot_NDS], directboot))
-            return Load_ROMLoadError;
+        char ext[5] = {0}; int _len = strlen(ROMPath[ROMSlot_NDS]);
+        strncpy(ext, ROMPath[ROMSlot_NDS] + _len - 4, 4);
+
+        if(!strncmp(ext, ".nds", 4) || !strncmp(ext, ".srl", 4) || !strncmp(ext, ".dsi", 4))
+        {
+            SetupSRAMPath(0);
+            if (!NDS::LoadROM(ROMPath[ROMSlot_NDS], SRAMPath[ROMSlot_NDS], directboot))
+                return Load_ROMLoadError;
+        }
+        else
+        {
+            u8 *romdata = nullptr; u32 romlen;
+            char romfilename[1024] = {0}, sramfilename[1024];
+            strncpy(sramfilename, SRAMPath[ROMSlot_NDS], 1024); // Use existing SRAMPath
+
+            int pos = strlen(sramfilename) - 1;
+            while(pos > 0 && sramfilename[pos] != '/' && sramfilename[pos] != '\\')
```

Comment:
I attempted using strrchr and got crashes on windows. This happens whenever sramfilename contains both '/' and '\' . For example - "C:/Users/Abcdjdj/Desktop\Pokemon Platinum.sav" In this case, it simply takes the last '/' and proceeds. I can add some code to take whichever of the 2 characters appears last but this seems unnecessary as the older code handles this case (and all other cases) just fine. If you don't mind, I'd like to revert back to the old code and skip using strrchr.

Reply:
sure. Ideally we would replace the entire cumbersome C string path code with one using std::filesystem, but that's a story for another time (which has it's own complications because this somehow would mean we couldn't support older macOS versions).

Reply:
Thanks, I've reverted it.